src/main/java/io/phasetwo/keycloak/magic/auth/EmailOtpAuthenticator.java

```diff
@@ -94,4 +112,26 @@ public void setRequiredActions(KeycloakSession session, RealmModel realm, UserMo
 
   @Override
   public void close() {}
+
+  public static String getAttemptedUsername(AuthenticationFlowContext context) {
```

Comment:
Seems like we can make this a utility method that both classes can use

Reply:
Done